apps/partners-reference/lib/formatApplicationData.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  ApplicationUpdate,
+  ApplicantUpdate,
+  Language,
+  IncomePeriod,
+  ApplicationSubmissionType,
+  ApplicationStatus,
+  Address,
+} from "@bloom-housing/core"
+import moment from "moment"
+
+/*
+  Some of fields are optional, not active, so it occurs 'undefined' as value.
+  This function eliminates those fields and parse to a proper format.
+*/
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+
+function getAddress(condition, addressData) {
+  const blankAddress: Address = {
+    id: undefined,
+    createdAt: undefined,
+    updatedAt: undefined,
+    street: "",
+    street2: "",
+    city: "",
+    state: "",
+    zipCode: "",
+  }
+
+  return condition ? addressData : blankAddress
+}
+
+export const formatApplicationData = (data: any, listingId, editMode: boolean) => {
+  const language: Language = data.application.language ? data.application.language : null
+
+  const submissionDate: Date | null = (() => {
+    // rename default (wrong property names)
+    const { birthDay: submissionDay, birthMonth: submissionMonth, birthYear: submissionYear } =
+      data.dateSubmitted || {}
+    const { hours, minutes = 0, seconds = 0, time } = data.timeSubmitted || {}
+
+    if (!submissionDay || !submissionMonth || !submissionYear) return null
+
+    const date = new Date()
+
+    date.setUTCDate(submissionDay)
+    date.setUTCMonth(submissionMonth - 1)
+    date.setUTCFullYear(submissionYear)
+
+    if (hours && minutes && seconds && time) {
+      const hourNumber = parseInt(hours, 10)
+      const hour24Clock = time === "am" ? hourNumber : hourNumber + 12
+      date.setUTCHours(hour24Clock, minutes, seconds)
+    } else {
+      date.setUTCHours(0, 0, 0, 0)
+    }
+
+    return date
+  })()
+
+  // create applicant
+  const applicant = ((): ApplicantUpdate => {
+    const { phoneNumber } = data
+    const { applicant: applicantData } = data.application
+    const phoneNumberType = applicantData.phoneNumberType ? applicantData.phoneNumberType : null
```

Comment:
1. Personally I didn't use it before but for such assignments there is a nullish coalescing operator in typescript 3.7 https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing
2. Type should be `string | null`